solidity/contracts/lib/zeto_common.sol

```diff
@@ -43,9 +43,17 @@ abstract contract ZetoCommon is OwnableUpgradeable {
     // should be called by escrow contracts that will use uploaded proofs
     // to execute transactions, in order to prevent the proof from being used
     // by parties other than the escrow contract
-    function lockProof(Commonlib.Proof calldata proof) public {
+    function lockProof(
+        Commonlib.Proof calldata proof,
+        address delegate
+    ) public {
         bytes32 proofHash = Commonlib.getProofHash(proof);
-        lockedProofs[proofHash] = msg.sender;
+        require(lockedProofs[proofHash] == address(0), "Proof already locked");
+        if (delegate != address(0)) {
+            lockedProofs[proofHash] = delegate;
+        } else {
+            lockedProofs[proofHash] = msg.sender;
```

Comment:
This is an interesting "default" behavior, but I wonder if it's what we want. Some contracts use "approve(0)" as a way to undo the approval (ie this can now be used by anyone). I think it makes sense to always pass in the delegate address, even if it is yourself.